src/nrnoc/multicore.cpp

```diff
@@ -284,33 +277,17 @@ static void wait_for_workers_timeit() {
 }
 
 static void send_job_to_slave(int i, void* (*job)(NrnThread*) ) {
-#if PERMANENT
-    pthread_mutex_lock(mut + i);
-    wc[i].job = job;
-    wc[i].flag = 1;
-    pthread_cond_signal(cond + i);
-    pthread_mutex_unlock(mut + i);
-#else
-    pthread_create(slave_threads + i, nullptr, (void* (*) (void*) ) job, (void*) (nrn_threads + i));
-#endif
-}
-
-void setaffinity(int i) {
-    int mask;
-    return;
-#if 0
-	cpu_set_t mask;
-	CPU_ZERO(&mask);
-	CPU_SET(i, &mask);
-	mask = (1 << i);
-	sched_setaffinity(0, 4, &mask);
-#endif
+    {
+        std::lock_guard<std::mutex> _{mut[i]};
```

Comment:
what do you about putting mut and cond inside wc? next PR or bad idea?

Reply:
Not sure... It's a bit awkward to use the cacheline alloc function with non-trivial types, and even if we wrap `nrn_cacheline_alloc` and `free` in a custom allocator then there is no convenient `std::allocate_unique` function to make a unique_ptr array using that (and `std::vector` is no good because the struct is not moveable). Maybe it's best left alone for now.